node/lib/util/status.js

```diff
@@ -43,235 +53,411 @@ const SubmoduleUtil       = require("../util/submodule_util");
+     * `cwd`.
+     *
+     * @param {Function} color
+     * @return {String}
+     */
+    print(color, cwd) {
+        let result = "";
+        const FILESTATUS = RepoStatus.FILESTATUS;
+        switch(this.status) {
             case FILESTATUS.ADDED:
-                return "new file:     ";
+                result += "new file:     ";
+                break;
             case FILESTATUS.MODIFIED:
-                return "modified:     ";
+                result += "modified:     ";
+                break;
             case FILESTATUS.REMOVED:
-                return "deleted:      ";
+                result += "deleted:      ";
+                break;
             case FILESTATUS.CONFLICTED:
-                return "conflicted:   ";
+                result += "conflicted:   ";
+                break;
             case FILESTATUS.RENAMED:
-                return "renamed:      ";
+                result += "renamed:      ";
+                break;
             case FILESTATUS.TYPECHANGED:
-                return "type changed: ";
+                result += "type changed: ";
+                break;
         }
+        result += path.relative(cwd, this.path);
+        result = color(result);
+        if ("" !== this.detail) {
+            result += ` (${this.detail})`;
+        }
+        return result;
     }
-    const innerIndent = "\t";
+}
 
-    // Print status of staged files first.
+exports.StatusDescriptor = StatusDescriptor;
 
-    if (0 !== Object.keys(repoStatus.staged).length) {
-        if ("" !== result) {
-            result += "\n";
-        }
-        result += "Changes staged to be commited:\n\n";
-        Object.keys(repoStatus.staged).sort().forEach(fileName => {
-            result += innerIndent;
-            const status = repoStatus.staged[fileName];
-            result += colors.green(statusDescription(status));
-            result += colors.green(fileName);
-            result += "\n";
-        });
+/**
+ * Return the specified `descriptors` sorted by path.
+ *
+ * @param {StatusDescriptor []} descriptors
+ * @return {StatusDescriptor []}
+ */
+exports.sortDescriptorsByPath = function (descriptors) {
+    return descriptors.sort((l, r) => {
+        const lPath = l.path;
+        const rPath = r.path;
+        return lPath === rPath ? 0 : (lPath < rPath ? -1 : 1);
+    });
+};
+
+/**
+ * Return a string describing the specified `statuses`, using the specified
+ * `color` function to apply color, printing paths relative to the specified
+ * `cwd`.
+ *
+ * @param {StatusDescriptor []} statuses
+ * @param {Function}            color
+ * @return {String}
+ */
+exports.printStatusDescriptors = function (statuses, color, cwd) {
+    assert.isArray(statuses);
+    assert.isFunction(color);
+
+    const sorted = exports.sortDescriptorsByPath(statuses);
+    const lines = sorted.map(status => "\t" + status.print(color, cwd));
+    if (0 !== lines.length) {
```

Comment:
you could move this check to the beginning of the function -- if (statuses.length == 0) { return ""; }